src/shared/elf-util.c

```diff
@@ -788,11 +789,15 @@ int parse_elf_object(int fd, const char *executable, bool fork_disable_dump, cha
          * bound since the core files have an upper size limit. It's also not doing any
          * system call or interacting with the system in any way, besides reading from
          * the file descriptor and writing into these four pipes. */
+        ForkFlags fork_flags = FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_WAIT|FORK_REOPEN_LOG;
+        if (userns_supported())
+                fork_flags |= FORK_NEW_USERNS|FORK_NEW_MOUNTNS|FORK_MOUNTNS_SLAVE;
```

Comment:
Hmm, the mountns flags should stay in, that's a different thing. Only turn off userns

Reply:
(i.e. depending on result of userns_supported() hence keep th caps we need early on, and then drop them hereafter for good)

Reply:
I wrote some explanation in the commit message of why I removed FORK_NEW_MOUNTNS as well:
```
Using a new mount namespace (CLONE_NEWNS) is also not possible without CONFIG_USER_NS,
since capabilities are dropped in the function submit_coredump() and using CLONE_NEWNS
alone thus results in an "operation not permitted" error.
```

Reply:
yes please keep CAP_SYS_ADMIN if we have no userns support until this point, and then drop it here (both in child and in parent)

Reply:
What is the advantage of keeping FORK_NEW_MOUNTNS and moving it to the code where CAP_SYS_ADMIN is still available?
Is it not equivalent from a security point of view to be in a situation where FORK_USERNS is not applied and the CAP_SYS_ADMIN capability is dropped? The mount points of the parent process will be visible inside the child process in both cases.

The issue of changing the logic dropping CAP_SYS_ADMIN capability is that it would make it much more wider change than the currently proposed patch (it would impact several systemd binaries), which I'm not sure how to test properly for regressions.